src/fu-engine.c

```diff
@@ -1054,17 +1044,83 @@ fu_engine_check_hardware_requirement (FuEngine *self, AsApp *app, GError **error
 }
 #endif
 
+static GHashTable *fu_engine_get_report_metadata (FuEngine *self);
+
+static gchar *
+fu_engine_requirement_convert_key (const gchar *name)
+{
+	struct {
+		const gchar *old;
+		const gchar *new;
+	} project_names[] = {
+		{ "FwupdVersion",		"org.freedesktop.fwupd" },
+		{ "FwupdateVersion",		"com.redhat.fwupdate" }, //FIXME?
+		{ "EfivarVersion",		"com.redhat.efivar" }, //FIXME?
+		{ "AppstreamGlibVersion",	"org.freedesktop.appstream-glib" },
```

Comment:
This one i'm also a little concerned about, there's a pretty big difference in a lot of fwupd code around compile time appstream glib version and runtime version.  You'll probably need to distinguish the two separately as a result.

Reply:
You could say the same about all of the versions there (with the exception of fwupd I think)

Reply:
True, so really all of these it's probably "safest" to err on compile time version unless they only provide runtime version information.

Reply:
I think the opposite; in both the libgusb and libfwupdate bugs mentioned they would be fixed by updating the relevant library.

Reply:
Won't the fwupdate-efi be kept in lockstep with the libfwupdate version in most cases?

Reply:
For deb/rpm distros yeah no problem.  It's the others I was concerned with.